backend/api/Services/BlobService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Api.Options;
+using Azure;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Extensions.Options;
+namespace Api.Services
+{
+    public interface IBlobService
+    {
+        public Task<byte[]> DownloadBlob(string blobName, string containerName, string accountName);
+
+        public AsyncPageable<BlobItem> FetchAllBlobs(string containerName, string accountName);
+    }
+
+    public class BlobService : IBlobService
+    {
+        private readonly IOptions<AzureAdOptions> _azureOptions;
+        private readonly ILogger<BlobService> _logger;
+
+        public BlobService(ILogger<BlobService> logger, IOptions<AzureAdOptions> azureOptions)
+        {
+            _logger = logger;
+            _azureOptions = azureOptions;
+        }
+
+        public async Task<byte[]> DownloadBlob(string blobName, string containerName, string accountName)
+        {
+            var blobContainerClient = GetBlobContainerClient(containerName, accountName);
+            var blobClient = blobContainerClient.GetBlobClient(blobName);
+
+            using var memoryStream = new MemoryStream();
+            await blobClient.DownloadToAsync(memoryStream);
+
+            return memoryStream.ToArray();
+        }
+
+        public AsyncPageable<BlobItem> FetchAllBlobs(string containerName, string accountName)
+        {
+            var blobContainerClient = GetBlobContainerClient(containerName, accountName);
+
+            AsyncPageable<BlobItem> blobs;
+            try
+            {
+                blobs = blobContainerClient.GetBlobsAsync(BlobTraits.Metadata);
+            }
+            catch (RequestFailedException e)
+            {
+                string errorMessage = $"Failed to fetch blob items because: {e.Message}";
+                _logger.LogError(e, "{ErrorMessage}", errorMessage);
+                throw;
+            }
+
+            return blobs;
+        }
```

Comment:
```suggestion
        public AsyncPageable<BlobItem> FetchAllBlobs(string containerName, string accountName)
        {
            try
            {
                var blobContainerClient = GetBlobContainerClient(containerName, accountName);
                return blobContainerClient.GetBlobsAsync(BlobTraits.Metadata);
            }
            catch (RequestFailedException e)
            {
                string errorMessage = $"Failed to fetch blob items because: {e.Message}";
                _logger.LogError(e, "{ErrorMessage}", errorMessage);
                throw;
            }
        }
```